test/infamy/netns.py

```diff
@@ -63,6 +63,21 @@ def popen(self, *args, **kwargs):
         args, kwargs = self._mangle_subprocess_args(args, kwargs)
         return subprocess.Popen(*args, **kwargs)
 
-    def runsh(self, script):
+    def runsh(self, script, *args, **kwargs):
         return self.run("/bin/sh", text=True, input=script,
-                        stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+                        stdout=subprocess.PIPE, stderr=subprocess.STDOUT, *args, **kwargs)
+    
+    def addip(self, addr, subnet=24):
+        self.runsh(f"""
+            set -ex 
+            ip link set iface up
+            ip addr add {addr}/{subnet} dev iface
+            """, check=True)
+    
+    def ping(self, daddr, count=1, timeout=2):
+        self.runsh(f"""set -ex; ping -c {count} -w {timeout} {daddr}""", check=True)
+
+    def ping_neg(self, daddr, count=1, timeout=2):
+        res = self.runsh(f"""set -ex; ping -c {count} -w {timeout} {daddr}""", check=False)
+        return res.returncode
```

Comment:
What if we had three ping methods:

- `ping()`: Runs with `check=False`, and just returns the `subprocess` result verbatim. Can be used to test if an address can be reached or not.
- `must_reach()`: Calls `ping()` and asserts that the returncode is 0, otherwise raises an exception. This is used in cases when a test should unconditionally fail if the address can't be reached.
- `must_not_reach()`: Calls `ping()` and asserts that the returncode is __not__ 0, otherwise raises an exception. This is used in cases when a test should unconditionally fail if the address __can__ be reached.